gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java

```diff
@@ -50,48 +54,78 @@
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
 import org.joda.time.Duration;
 
 import java.io.IOException;
 import java.util.Set;
 import java.util.concurrent.Future;
-
-import autovalue.shaded.com.google.common.common.collect.Sets;
-import io.grpc.Status.Code;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 
 public class DefaultPubSubRpc implements PubSubRpc {
 
   private final PublisherApi publisherApi;
   private final SubscriberApi subscriberApi;
+  private final ScheduledExecutorService executor =
+      MoreExecutors.getExitingScheduledExecutorService(new ScheduledThreadPoolExecutor(8));
 
   public DefaultPubSubRpc(PubSubOptions options) throws IOException {
     try {
-      // Provide (and use a common thread-pool).
-      // This depends on https://github.com/googleapis/gax-java/issues/73
-      PublisherSettings.Builder pbuilder =
-          PublisherSettings.defaultBuilder()
-              .provideChannelWith(options.authCredentials().credentials())
-              .applyToAllApiMethods(apiCallSettings(options));
-      publisherApi = PublisherApi.create(pbuilder.build());
-      SubscriberSettings.Builder sBuilder =
-          SubscriberSettings.defaultBuilder()
-              .provideChannelWith(options.authCredentials().credentials())
-              .applyToAllApiMethods(apiCallSettings(options));
-      subscriberApi = SubscriberApi.create(sBuilder.build());
+      PublisherSettings.Builder pubBuilder =
+          PublisherSettings.defaultBuilder().provideExecutorWith(executor, false);
+      SubscriberSettings.Builder subBuilder =
+          SubscriberSettings.defaultBuilder().provideExecutorWith(executor, false);
+      // todo(mziccard): PublisherSettings should support null/absent credentials for testing
+      if (options.host().contains("localhost")
+          || options.authCredentials().equals(AuthCredentials.noAuth())) {
+        ManagedChannel channel = NettyChannelBuilder.forTarget(options.host())
+            .negotiationType(NegotiationType.PLAINTEXT)
+            .build();
+        pubBuilder.provideChannelWith(channel, true);
+        subBuilder.provideChannelWith(channel, true);
+      } else {
+        GoogleCredentials credentials = options.authCredentials().credentials();
+        pubBuilder.provideChannelWith(
+            credentials.createScoped(PublisherSettings.DEFAULT_SERVICE_SCOPES));
+        subBuilder.provideChannelWith(
+            credentials.createScoped(SubscriberSettings.DEFAULT_SERVICE_SCOPES));
+      }
+      pubBuilder.applyToAllApiMethods(apiCallSettings(options));
+      subBuilder.applyToAllApiMethods(apiCallSettings(options));
+      publisherApi = PublisherApi.create(pubBuilder.build());
+      subscriberApi = SubscriberApi.create(subBuilder.build());
     } catch (Exception ex) {
       throw new IOException(ex);
     }
   }
 
+  private static long translateTimeout(long timeout) {
```

Comment:
Worth thinking how to make this logic common for all services (which can override a different default value).